### PR TITLE
LOG-4072: refactor CR reconciliation

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* refactors loading, migration, validation of CRs into common code used by the controllers

### Links
* https://issues.redhat.com/browse/LOG-4072

cc @Clee2691

### CODE REVIEW

File: internal/runtime/logging.go [hunk 1]
```diff
@@ -10,3 +10,10 @@ func NewClusterLogForwarder(namespace, name string) *loggingv1.ClusterLogForward
 	Initialize(clf, namespace, name)
 	return clf
 }
+
+// NewClusterLogging returns a ClusterLogging with default name and namespace.
+func NewClusterLogging(namespace, name string) *loggingv1.ClusterLogging {
+	clf := &loggingv1.ClusterLogging{}
```
Comment (Clee2691) [inline internal/runtime/logging.go:16]: variable should be `cl` instead of `clf`.

**Comment by cahartma** [comment]
No other comments from me, but has been reviewed and tested locally.   lgtm once updated with small change above.

**Pull request merged**